### PR TITLE
Update github.com/gardener/etcd-druid

### PR DESCRIPTION
**Pull request by ialidzhikov**
from v0.12.3 to v0.15.0

**How to categorize this PR?**

/kind enhancement

**What this PR does / why we need it**:
This PR updates `github.com/gardener/etcd-druid` dependencies and CRDs. This is prerequisite for https://github.com/gardener/gardener/issues/6718 to be able to consume https://github.com/gardener/etcd-druid/pull/438.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/6718

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by ialidzhikov** [comment]
/hold
I will add a second commit that updates the CRDs but the CRDs that I found in the upstream did not have the newly introduced fields. I need to clarify this first with @shreyas-s-rao.

File: pkg/operation/botanist/component/etcd/crds/templates/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Source: https://github.com/gardener/etcd-druid/blob/v0.15.0/charts/druid/charts/crds/templates/10-crd-druid.gardener.cloud_etcdcopybackupstasks.yaml
```
Comment (ialidzhikov) [inline pkg/operation/botanist/component/etcd/crds/templates/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml:1]: The changes in this file should be caused only by changes in the indentation.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**